x-pack/plugins/maps/public/components/map/feature_tooltip.js

```diff
@@ -5,12 +5,77 @@
  */
 
 import React, { Fragment } from 'react';
-import { EuiFlexGroup, EuiFlexItem, EuiButtonIcon } from '@elastic/eui';
+import { EuiFlexGroup, EuiFlexItem, EuiButtonIcon, EuiCallOut, EuiLoadingSpinner } from '@elastic/eui';
 import { i18n } from '@kbn/i18n';
 
 
 export class FeatureTooltip extends React.Component {
 
+  state = {
+    properties: undefined,
+    loadPropertiesErrorMsg: undefined,
+  };
+
+  componentDidMount() {
+    this._isMounted = true;
+    this.prevLayerId = undefined;
```

Comment:
Putting `prevLayerId` and `prevFeatureId` in state would require `_fetchProperties` to be more complex than it currently is. `setState` is async so there is guarantee that state has been updated before the check to ignore old requests occurs. By putting `prevLayerId` and `prevFeatureId` on the instance, `_fetchProperties` can set the values synchronously and avoid potential timing issues. `prevLayerId` and `prevFeatureId` do not need to be in state since the values are not used for rendering.